misc/build-wyrm-transpiler.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+set -ex
+source common.sh
+
+VERSION=$1
+if [[ "${VERSION}" = "trunk" ]]; then
+    VERSION=trunk-$(date +%Y%m%d)
+    BRANCH=main
+else
+    BRANCH=V${VERSION}
+fi
+
+URL=https://github.com/jeremy-rifkin/wyrm.git
+
+FULLNAME=wyrm-${VERSION}.tar.xz
+OUTPUT=$2/${FULLNAME}
+
+REVISION="wyrm-$(get_remote_revision "${URL}" "heads/${BRANCH}")"
+LAST_REVISION="${3:-}"
+
+initialise "${REVISION}" "${OUTPUT}" "${LAST_REVISION}"
+
+PREFIX=$(pwd)/prefix
+DIR=$(pwd)/wyrm
+
+git clone "${URL}" "${DIR}"
+
+cd "${DIR}/transpiler"
+
+mkdir build
+cd build
+export CXX=/opt/compiler-explorer/gcc-12.1.0/bin/g++
+export CC=/opt/compiler-explorer/gcc-12.1.0/bin/gcc
+cmake .. -GNinja -DCMAKE_BUILD_TYPE=Debug
+ninja
+
+mkdir -p "${PREFIX}"
+cd ..
+mv build "${PREFIX}/build"
```

Comment:
Do we want the tarball to contain this `build` subdirectory? I _think_ this will keep that dir in $PREFIX and I'd expect us not to want the `build/` directory there and also all the contents (like all the intermediate CMake files etc too? though I imagine they're not too big?) In other cmake-based setups we use a `ninja install` step or whatever, but it does rely on the upstream having set up installation properly :)

Reply:
Ah a ninja install is a good idea. I looked at the contents of the tarball briefly and figured "well I see the .so somewhere so lgtm!" but this is a good point